### ISSUE TITLE
Rule Proposal: max-lines

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#6078]
From [maximumNumberOfLines].

This rule will disallow that a file gets the number of lines more than the specific integer.

```js
{
    "max-lines": ["error", {
        "max": 300,
        "skipBlankLines": false,
        "skipComments": false
    }]
}
```

- `max` (default is `300`) - The maximum count of lines per file.
- `skipBlankLines` (default is `false`) - If this is `true`, this rule skips blank lines to count lines.
- `skipComments` (default is `false`) - If this is `true`, this rule skips comment lines to count lines.

[maximumNumberOfLines]: http://jscs.info/rule/maximumNumberOfLines

**Comment by alberto** [comment]
I'm a champion

**Comment by ilyavolodin** [comment]
This rule is very close to complexity (no direct correlation, but never the less more lines usually means higher complexity). Do we need this in the core or should it go into a plugin of some kind?

**Comment by alberto** [comment]
@ilyavolodin there is a bit correlation, but they are quite different. Also, `complexity` is per function, same as `max-statements`

**Comment by doberkofler** [comment]
+1

**Comment by alberto** [comment]
@eslint/eslint-team opinions?

**Comment by mikesherov** [comment]
:+1:

**Comment by zxqfox** [comment]
👍

**Comment by nzakas** [comment]
:+1:

**Comment by platinumazure** [comment]
For `skipComments`, I assume that refers to lines with only comments on them?

Lines I assume count under that definition:

```js
// comment

/* comment */

/* comment one */ /* comment two */

/* comment one */ // comment two
```

Lines I assume don't count under that definition:

```js
var foo = "bar"; // comment
```

And I'm happy to take this on if @alberto isn't super attached.
